proto/attestation/endorsement.proto

```diff
@@ -43,42 +43,25 @@ message RootLayerEndorsements {
   TransparentReleaseEndorsement stage0_endorsement = 2;
 };
 
-message LinuxKernelEndorsement {
-  TransparentReleaseEndorsement kernel_binary = 1;
-  TransparentReleaseEndorsement kernel_cmd_line = 2;
-  TransparentReleaseEndorsement acpi = 3;
-}
-
-message OakRestrictedKernelEndorsement {
-  TransparentReleaseEndorsement oak_restricted_kernel = 1;
+message KernelLayerEndorsements {
+  oneof type {
+    TransparentReleaseEndorsement linux_kernel = 1;
+    TransparentReleaseEndorsement oak_restricted_kernel = 2;
+  }
 }
 
-message CborFieldsEndorsement {
-  // Map from an OpenDICE binary label to a Transparent Release endorsement.
-  map<int32, TransparentReleaseEndorsement> endorsements = 1;
+message CustomLayerEndorsements {
+  TransparentReleaseEndorsement binary = 1;
 }
 
-message LayerEndorsements {
-  oneof layer_type {
-    LinuxKernelEndorsement linux_kernel = 1;
-
-    OakRestrictedKernelEndorsement oak_restricted_kernel = 2;
-
-    TransparentReleaseEndorsement generic_binary = 3;
-
-    // Ideally, we would only use the concrete endorsements and make this
-    // obsolete. But it should remain here so no one is blocked.
-    CborFieldsEndorsement cbor_fields = 4;
-  }
-};
-
 // Endorsement contains statements that some entity (e.g. a hardware provider)
 // vouches for the integrity of claims about the TEE or the software running
 // on it.
 //
 // The name is chosen to match the RATS terminology:
 // <https://www.rfc-editor.org/rfc/rfc9334.html#name-endorsements>
 message Endorsements {
-  RootLayerEndorsements root_layer = 1;
-  repeated LayerEndorsements layers = 2;
+  RootLayerEndorsements root_layer = 1;                // Layer 0
+  KernelLayerEndorsements kernel_layer = 2;            // Layer 1
+  repeated CustomLayerEndorsements custom_layers = 3;  // Layers 2 through N
```

Comment:
I wonder if (maybe at some point in the future) it would be easier to just define separate full chains (one for Oak Restricted Kernel, one for Oak Containers, one for other stacks), with concrete static types, then *maybe* also a generic one for other unforeseen use cases.

@conradgrobler @ipetr0v @thmsbinder WDYT?

Reply:
I assume the idea would be to mirror that in the evidence as well. I have mixed feelings about that. I like that it is more explicit, and will make the verification code clearer. But I also think it will be a bit less convenient to generate the DICE chain.

Overall the increase clarity of verification logic probably makes up for the more complex generation code, so probably worth it.

Reply:
Yes, the evidence wasn't suggesting for mirroring the kernel vs. custom layer structure. But same with Endorsement, it is not more than one (or more?) TransparentReleaseEndorsement, with ordering. The purpose of KernelLayerEndorsements is just to refer to the endorsements by name and not by index.

Since there is no protocol buffer inheritance, we would have several top-level verification functions which I find undesirable. Only a single top-level verification function which corresponds to the top-level proto buffers is nicer. Immediately below that we can have concrete messages for the types of layers.

A lot depends on the variability in the layers w.r.t. the protos, I am not fully clear about what to expect here in the future.

Reply:
@ipetr0v I think we are lacking a special layer type for Layer 2, with two endorsements and two binaries. Question is, does it depend on Layer 1 (Linux or Oak Resticted Kernel)?

Reply:
> @ipetr0v I think we are lacking a special layer type for Layer 2, with two endorsements and two binaries (Linux distro, orchestrator). This assumes that they are separate standalone binaries - is this correct? Question is, does it depend on Layer 1 (whether it contains Linux or Oak Restricted Kernel)?
> 
> But the scheme is roughly clear.

After a discussion with @conradgrobler: for Layer 2 we will have a single `.tar` file that contains both the Linux distribution and the Orchestrator code. Though it will may be different for other use-cases.